xgcm/grid.py

```diff
@@ -758,6 +765,122 @@ def max(
             keep_coords,
         )
 
+    def transform(
+        self,
+        da,
+        target,
+        target_data=None,
+        target_dim=None,
+        method="linear",
+        mask_edges=False,
+    ):
+        """Transform data along a 1D axis
+
+        Parameters
+        ----------
+        da : xr.Dataarray
+            Input data
+        target : {np.array, xr.Dataarray}
+            Target data for transformation. Dependin on the method is interpreted as cell center (method='linear') or cell bounds (method='conservative).
+            When passed as numpy array the resulting dimension is named according to `target_data`, if provided as xr.Dataarray naming is inferred from dimensions.
+        target_data : xr.Datarray, optional
+            Optional data to transform onto (e.g. a tracer like density or temperature). Defaults to None, which infers the appropriate coordinate along `axis` (e.g. the depth).
```

Comment:
I find this very confusing language, but I'm struggling to come up with something better.